teerex/src/lib.rs

```diff
@@ -178,33 +175,26 @@ decl_module! {
 			Ok(())
 		}
 
-		// the integritee-service calls this function for every processed call to confirm a state update
-		#[weight = (<T as Config>::WeightInfo::confirm_call(), DispatchClass::Normal, Pays::Yes)]
-		pub fn confirm_call(origin, shard: ShardIdentifier, call_hash: H256, ipfs_hash: Vec<u8>) -> DispatchResult {
+		// the integritee worker calls this function for every processed parentchainblock to confirm a state update
+		#[weight = (<T as Config>::WeightInfo::confirm_processed_parentchainblock(), DispatchClass::Normal, Pays::Yes)]
+		pub fn confirm_processed_parentchainblock(origin, block_hash: H256, trusted_calls_merkle_root: H256) -> DispatchResult {
 			let sender = ensure_signed(origin)?;
 			Self::is_registered_enclave(&sender)?;
-			let sender_index = Self::enclave_index(&sender);
-			ensure!(<EnclaveRegistry::<T>>::get(sender_index).mr_enclave.encode() == shard.encode(), <Error<T>>::WrongMrenclaveForShard);
-			<LatestIpfsHash>::insert(shard, ipfs_hash.clone());
-			<WorkerForShard>::insert(shard, sender_index);
-			log::debug!("call confirmed with shard {:?}, call hash {:?}, ipfs_hash {:?}", shard, call_hash, ipfs_hash);
-			Self::deposit_event(RawEvent::CallConfirmed(sender, call_hash));
-			Self::deposit_event(RawEvent::UpdatedIpfsHash(shard, sender_index, ipfs_hash));
+			log::debug!("Processed parentchain block confirmed for mrenclave {:?}, block hash {:?}", sender, block_hash);
+			Self::deposit_event(RawEvent::ProcessedParentchainBlock(sender, block_hash, trusted_calls_merkle_root));
 			Ok(())
 		}
 
-		// the integritee-service calls this function for every processed block to confirm a state update
-		#[weight = (<T as Config>::WeightInfo::confirm_block(), DispatchClass::Normal, Pays::Yes)]
-		pub fn confirm_block(origin, shard: ShardIdentifier, block_hash: H256, ipfs_hash: Vec<u8>) -> DispatchResult {
+		// the integritee worker calls this function for every produced sidechainblock
```

Comment:
It should be 'proposed' sidechainblock to be consistent with the call's name. :)

Reply:
Totally agreed. Also made the comments a little bit more clangenb like :-)

Reply:
Haha, thanks, I feeld honored. :)